nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKudu.java

```diff
@@ -0,0 +1,199 @@
+            .name("KUDU Masters")
+            .description("List all kudu masters's ip with port (e.g. 7051), comma separated")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("Table Name")
+            .description("The name of the Kudu Table to put data into")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("record-reader")
+            .displayName("Record Reader")
+            .description("The service for reading records from incoming flow files.")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .build();
+
+    protected static final PropertyDescriptor SKIP_HEAD_LINE = new PropertyDescriptor.Builder()
+            .name("Skip head line")
+            .description("Set it to true if your first line is the header line e.g. column names")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is routed to this relationship after it has been successfully stored in Kudu")
+            .build();
+    protected static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be sent to Kudu")
+            .build();
+
+    public static final String RECORD_COUNT_ATTR = "record.count";
+
+    protected String kuduMasters;
+    protected String tableName;
+    protected boolean skipHeadLine;
+
+    protected KuduClient kuduClient = null;
+    protected KuduTable kuduTable = null;
+    protected KuduSession kuduSession = null;
+    protected RecordReaderFactory recordReaderFactory;
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        try {
+            tableName = context.getProperty(TABLE_NAME).getValue();
+            kuduMasters = context.getProperty(KUDU_MASTERS).getValue();
+            skipHeadLine = context.getProperty(SKIP_HEAD_LINE).asBoolean();
+
+            recordReaderFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+
+            kuduClient = (kuduClient == null) ? getKuduConnection(kuduMasters) : kuduClient;
+            kuduTable = (kuduTable == null) ? kuduClient.openTable(tableName) : kuduTable;
+            kuduSession = (kuduSession == null) ? kuduClient.newSession() : kuduSession;
```

Comment:
what if the session were to die while the processor was executing?  Do these objects represent connections?  If yes we probably need to do this object lazily/on-demand triggered during an actual process execution rather than onScheduled.  If not, and the connection were to die then the processor would require manual restarting.